packages/dds/tree/src/feature-libraries/forest-proxy/forestProxy.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { FieldKey } from "../../tree";
+import { IEditableForest, TreeNavigationResult, ITreeSubscriptionCursor } from "../../forest";
+
+export const proxySymbol = Symbol("forest-proxy");
+
+class TargetForest {
+	private readonly _cursor: ITreeSubscriptionCursor;
+
+	constructor(
+		public readonly forest: IEditableForest,
+		_cursor?: ITreeSubscriptionCursor,
+	) {
+		if (_cursor) {
+			this._cursor = _cursor.fork();
+		} else {
+			this._cursor = forest.allocateCursor();
+			forest.tryMoveCursorTo(forest.root(forest.rootField), this._cursor);
```

Comment:
this being the only use of forest in here makes me think maybe I should change the ITreeSubscriptionCursor API a bit so you don't need top reference the forest at all. I should follow up on this.

Reply:
Another option would be to make it mandatory to pass a cursor that is positioned at the root of the subtree you want to proxify.

However, we'll may still need a reference to forest in order to apply edits (unless we want to pass a callback for apply, like `apply: (Delta.Root) => void`).

Reply:
> this being the only use of forest in here makes me think maybe I should change the ITreeSubscriptionCursor API a bit so you don't need top reference the forest at all. I should follow up on this.

I would keep a forest as an entity, which is clear for the "user", whereas cursors for me are kind of implementation details (at least, in this particular case). On the other hand, it would be nice to have a public reference to the forest the cursor attached to, so that this code could be simplified plus the comment from @DLehenbauer regarding edits.

> Another option would be to make it mandatory to pass a cursor that is positioned at the root of the subtree you want to proxify.

This usecase seems for me to happen quite often ("allocate a cursor and move it to the root"), especially at the very beginning of each and every development in case of new "customers", so why not to add it to the forest API?